fe/fe-core/src/main/java/com/starrocks/system/SystemInfoService.java

```diff
@@ -291,6 +294,46 @@ public ShowResultSet modifyBackendHost(ModifyBackendClause modifyBackendClause)
         return new ShowResultSet(builder.build(), messageResult);
     }
 
+
+    public ShowResultSet modifyComputeNodeProperty(ModifyComputeNodeClause modifyComputeNodeClause) throws DdlException {
+        String computeNodeHostPort = modifyComputeNodeClause.getComputeNodeHostPort();
+        Map<String, String> properties = modifyComputeNodeClause.getProperties();
+
+        // check compute node existence
+        ComputeNode computeNode = getComputeNodeWithHeartbeatPort(computeNodeHostPort.split(":")[0],
+                Integer.parseInt(computeNodeHostPort.split(":")[1]));
+        if (null == computeNode) {
+            throw new DdlException(String.format("computeNode [%s] not found", computeNodeHostPort));
+        }
+
+        ShowResultSetMetaData.Builder builder = ShowResultSetMetaData.builder();
+        builder.addColumn(new Column("Message", ScalarType.createVarchar(1024)));
+        List<List<String>> messageResult = new ArrayList<>();
+
+        // update backend based on properties
+        for (Map.Entry<String, String> entry : properties.entrySet()) {
+            if (entry.getKey().equals(AlterSystemStmtAnalyzer.PROP_KEY_GROUP)) {
+                // "" means clean group label
+                if (entry.getValue().isEmpty()) {
+                    computeNode.setResourceIsolationGroup("");
+                    continue;
+                }
+                computeNode.setResourceIsolationGroup(getResourceIsolationGroupFromProperties(properties));
```

Comment:
Sure, the logging could could be useful, added that. It should be a noop to set the same resource group id, and it seems more robust to bugs to do it anyway, so I'll keep the call to set the group and persist the action to the metadata record regardless

Reply:
I'd be fine with that, but for future reference, could you elaborate a bit on why it's more robust to bugs if you perform the update regardless?

Reply:
Sure. The first thing I'd say is that I don't think there's a downside to repeating an operation like this. 
Possible upsides
1.  it could save us in a situation where the frontends fell out of sync: e.g. _this_ frontend, that is, the one handling the MODIFY COMPUTE NODE statement, has knows that the frontend belongs to the given compute node, but the _other_ frontends might have fallen out of sync. In that case, we'd want to persist the operation to the metadata log so the other frontends can pick up on it, and if we're going to persist it to the metadata log, I think it really makes sense to perform the operation
2. It's future proofing. It could save us in a situation where we've later added some side effect to `setResourceIsolationGroup` that we also want executed. Of course, we could change the behavior of this function at the time that we change `setResourceIsolationGroup` to add said side effect, but that could be an easy thing to miss.